### ISSUE TITLE
Can't run Astro 4.4.0 with Bun 1.0.27

### ISSUE THREAD

**Issue by SemionVino** [oven-sh/bun#8960]
### What version of Bun is running?

1.0.27+c34bbb2e3

### What platform is your computer?

Linux 5.15.133.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

After running both `bunx --bun astro dev` and `bunx astro dev` and heading to `localhost:4321` I'm getting the following error:
```
 astro  v4.4.0 ready in 206 ms

┃ Local    http://localhost:4321/
┃ Network  use --host to expose

17:56:30 watching for file changes...
17:56:35 [200] / 25ms
17:56:35 [ERROR] undefined is not a function
  Stack trace:
    at /root/songspectrum/songspectrum-astro/node_modules/astro/dist/vite-plugin-astro-server/response.js:52:50
    [...] See full stack trace in the browser, or rerun with --verbose.
An unexpected error occurred in the middle of the stream. 47 |     _headers["set-cookie"] = headers.getSetCookie();
48 |   }
49 |   res.writeHead(status, _headers);
50 |   if (body) {
51 |     if (Symbol.for("astro.responseBody") in webResponse) {
52 |       let stream = webResponse[Symbol.for("astro.responseBody")];
                                ^
TypeError: undefined is not a function
      at pull (:9:15)
      at onPullDirectStream (:1:11)
      at /root/songspectrum/songspectrum-astro/node_modules/astro/dist/vite-plugin-astro-server/response.js:52:50

1 | (function (reason) {"use strict";
                        ^
TypeError: ReadableStreamDirectController is now closed
      at onReadableStreamDirectControllerClosed (:1:21)
1 | (function (reason) {"use strict";
                        ^
TypeError: ReadableStreamDirectController is now closed
      at onReadableStreamDirectControllerClosed (:1:21)
1 | (function (reason) {"use strict";
                        ^
TypeError: ReadableStreamDirectController is now closed
      at onReadableStreamDirectControllerClosed (:1:21)
1 | (function (reason) {"use strict";
                        ^
TypeError: ReadableStreamDirectController is now closed
      at onReadableStreamDirectControllerClosed (:1:21)
```

Downgrading Astro to 4.3.0 solves the issue.

### What is the expected behavior?

A rendered web page

### What do you see instead?

A broken, half rendered web page and the mentioned error in the console.

**Comment by Jarred-Sumner** [comment]
Looks like my fix didn’t fix it. That’s what we get for writing unit tests instead of integration tests.

**Comment by Jarred-Sumner** [comment]
@SemionVino how can I reproduce this? I'm unable to with a create astro app

**Comment by Jarred-Sumner** [comment]
Fixed in #8999
